test/integration/vpc_e2e_test.go

```diff
@@ -0,0 +1,144 @@
+	"antrea.io/nephe/test/utils"
+)
+
+var _ = Describe(fmt.Sprintf("%s,%s: Cloud Resource VPC", focusAws, focusAzure), func() {
+
+	var (
+		ns = staticVMNS
+	)
+
+	BeforeEach(func() {
+		logf.Log.Info("BeforeEach", "namespace", ns)
+	})
+
+	AfterEach(func() {
+		result := CurrentGinkgoTestDescription()
+		if result.Failed {
+			if len(supportBundleDir) > 0 {
+				logf.Log.Info("Collect support bundles for test failure.")
+				fileName := utils.GenerateNameFromText(result.FullTestText, testFocus)
+				utils.CollectSupportBundle(kubeCtl, path.Join(supportBundleDir, fileName), cloudVPC, withAgent)
+			}
+			if preserveSetupOnFail {
+				logf.Log.V(1).Info("Preserve setup on failure")
+				return
+			}
+		}
+
+		// cleanup the test configuration by deleting the namespace.
+		if ns == nil {
+			return
+		}
+		logf.Log.V(1).Info("Delete namespace", "ns", ns.Name)
+		err := k8sClient.Delete(context.TODO(), ns)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Basic Test for VPC", func() {
+		vmIDs := cloudVPC.GetVMs()
+		vpcID := cloudVPC.GetVPCID()
+		vpcName := cloudVPC.GetVPCName()
+		cloudProvider := strings.Split(cloudProviders, ",")[0]
+		kind := reflect.TypeOf(v1alpha1.VirtualMachine{}).Name()
+		// Create a Namespace
+		if len(ns.Status.Phase) == 0 {
+			// Create ns if not already present.
+			ns.Name = "test-cloud-e2e-" + fmt.Sprintf("%x", rand.Int())
+			ns.Labels = map[string]string{"name": ns.Name}
+			err := k8sClient.Create(context.TODO(), ns)
+			Expect(err).ToNot(HaveOccurred())
+			logf.Log.V(1).Info("Created namespace", "ns", ns.Name)
+		}
+
+		// Configure CPA.
+		logf.Log.V(1).Info("Configure and create CPA")
+		accountParams := cloudVPC.GetCloudAccountParameters("test-cloud-account"+ns.Name, ns.Name, cloudCluster)
+		err := utils.AddCloudAccount(kubeCtl, accountParams)
+		Expect(err).ToNot(HaveOccurred())
+
+		vpcObjectName := ""
+		switch cloudProvider {
+		case string(v1alpha1.AWSCloudProvider):
+			vpcObjectName = vpcID
+		case string(v1alpha1.AzureCloudProvider):
+			vpcObjectName = cloudUtils.GenerateShortResourceIdentifier(vpcID, vpcName)
+		default:
+			logf.Log.Error(nil, "Invalid provider", "Provider", cloudProvider)
+		}
+
+		// Verify vpc inventory.
+		logf.Log.V(1).Info("Get vpcID of VM Cluster", "vpcID", vpcID)
+		err = wait.Poll(time.Second*5, time.Second*30, func() (bool, error) {
+			cmd := fmt.Sprintf("get vpc -A -o json")
+			out, err := kubeCtl.Cmd(cmd)
+			Expect(out).To(ContainSubstring(vpcObjectName))
+			// Find if vpc with expected Id is present in the vpc inventory.
+			cmd = fmt.Sprintf("get vpc %v -n %v -o json -o=jsonpath={.spec.Id}", vpcObjectName, ns.Name)
+			out, err = kubeCtl.Cmd(cmd)
+			logf.Log.V(1).Info("Got VPC", "vpcId", out)
+			if err != nil {
+				logf.Log.V(1).Info("Failed to get vpc", "vpc object name", vpcObjectName, "err", err)
+				return false, nil
+			}
+			Expect(out).To(Equal(vpcID))
+			return true, nil
+		})
+		Expect(err).ToNot(HaveOccurred(), "timeout waiting to retrieve vpc")
```

Comment:
verify no VM is onboard yet

Reply:
added check to verify vms are not imported at this point.